Levando em consideração os ataques de tubarões ocorridos no século 21, nos países USA e Austrália quais seriam a/as praia/as, onde seria necessário um controle de acessos?

In [2]:
import pandas as pd
import seaborn as sns
import numpy as np

In [3]:
attacks =pd.read_csv('attacks.csv', sep=',', index_col = -1, encoding='latin-1')

In [4]:
attacks

Case Number         Date    Year        Type    Country             Area  \
NaN  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
NaN  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
NaN  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
NaN  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
NaN  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   
..          ...          ...     ...         ...        ...              ...   
NaN         NaN          NaN     NaN         NaN        NaN              NaN   
NaN         NaN          NaN     NaN         NaN        NaN              NaN   
NaN         NaN          NaN     NaN         NaN        NaN              NaN   
NaN         NaN          NaN     NaN         NaN        NaN              NaN   
NaN          xx          NaN     NaN         NaN        NaN              NaN   

                           Location     Activity             Name Sex   ...  \
NaN     Oceanside, San Diego County     Paddling      Julie Wolfe    F  ...   
NaN  St. Simon Island, Glynn County     Standing  Adyson McNeely     F  ...   
NaN                    Habush, Oahu      Surfing      John Denges    M  ...   
NaN              Arrawarra Headland      Surfing             male    M  ...   
NaN                        La Ticla  Free diving   Gustavo Ramos     M  ...   
..                              ...          ...              ...  ...  ...   
NaN                             NaN          NaN              NaN  NaN  ...   
NaN                             NaN          NaN              NaN  NaN  ...   
NaN                             NaN          NaN              NaN  NaN  ...   
NaN                             NaN          NaN              NaN  NaN  ...   
NaN                             NaN          NaN              NaN  NaN  ...   

              Time         Species           Investigator or Source  \
NaN          18h00      White shark                R. Collier, GSAF   
NaN  14h00  -15h00              NaN  K.McMurray, TrackingSharks.com   
NaN          07h45              NaN  K.McMurray, TrackingSharks.com   
NaN            NaN        2 m shark                  B. Myatt, GSAF   
NaN            NaN  Tiger shark, 3m                       A .Kipper   
..             ...              ...                             ...   
NaN            NaN              NaN                             NaN   
NaN            NaN              NaN                             NaN   
NaN            NaN              NaN                             NaN   
NaN            NaN              NaN                             NaN   
NaN            NaN              NaN                             NaN   

                          pdf  \
NaN      2018.06.25-Wolfe.pdf   
NaN    2018.06.18-McNeely.pdf   
NaN     2018.06.09-Denges.pdf   
NaN  2018.06.08-Arrawarra.pdf   
NaN      2018.06.04-Ramos.pdf   
..                        ...   
NaN                       NaN   
NaN                       NaN   
NaN                       NaN   
NaN                       NaN   
NaN                       NaN   

                                          href formula  \
NaN  http://sharkattackfile.net/spreadsheets/pdf_di...   
NaN  http://sharkattackfile.net/spreadsheets/pdf_di...   
NaN  http://sharkattackfile.net/spreadsheets/pdf_di...   
NaN  http://sharkattackfile.net/spreadsheets/pdf_di...   
NaN  http://sharkattackfile.net/spreadsheets/pdf_di...   
..                                                 ...   
NaN                                                NaN   
NaN                                                NaN   
NaN                                                NaN   
NaN                                                NaN   
NaN                                                NaN   

                                                  href Case Number.1  \
NaN  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
NaN

In [5]:
attacks.columns

Index(['Case Number', 'Date', 'Year', 'Type', 'Country', 'Area', 'Location',
       'Activity', 'Name', 'Sex ', 'Age', 'Injury', 'Fatal (Y/N)', 'Time',
       'Species ', 'Investigator or Source', 'pdf', 'href formula', 'href',
       'Case Number.1', 'Case Number.2', 'original order', 'Unnamed: 22'],
      dtype='object')

In [6]:
attacks.rename(columns={'Fatal (Y/N)':'fatal'}, inplace=True)


In [7]:
attacks.columns = [x.strip().lower().replace(' ','_') for x in attacks.columns]
attacks.columns

Index(['case_number', 'date', 'year', 'type', 'country', 'area', 'location',
       'activity', 'name', 'sex', 'age', 'injury', 'fatal', 'time', 'species',
       'investigator_or_source', 'pdf', 'href_formula', 'href',
       'case_number.1', 'case_number.2', 'original_order', 'unnamed:_22'],
      dtype='object')

In [8]:
attacks.dropna(how='all') 


case_number         date    year        type    country             area  \
NaN  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
NaN  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
NaN  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
NaN  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
NaN  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   
..          ...          ...     ...         ...        ...              ...   
NaN           0          NaN     NaN         NaN        NaN              NaN   
NaN           0          NaN     NaN         NaN        NaN              NaN   
NaN           0          NaN     NaN         NaN        NaN              NaN   
NaN           0          NaN     NaN         NaN        NaN              NaN   
NaN          xx          NaN     NaN         NaN        NaN              NaN   

                           location     activity             name  sex  ...  \
NaN     Oceanside, San Diego County     Paddling      Julie Wolfe    F  ...   
NaN  St. Simon Island, Glynn County     Standing  Adyson McNeely     F  ...   
NaN                    Habush, Oahu      Surfing      John Denges    M  ...   
NaN              Arrawarra Headland      Surfing             male    M  ...   
NaN                        La Ticla  Free diving   Gustavo Ramos     M  ...   
..                              ...          ...              ...  ...  ...   
NaN                             NaN          NaN              NaN  NaN  ...   
NaN                             NaN          NaN              NaN  NaN  ...   
NaN                             NaN          NaN              NaN  NaN  ...   
NaN                             NaN          NaN              NaN  NaN  ...   
NaN                             NaN          NaN              NaN  NaN  ...   

              time          species          investigator_or_source  \
NaN          18h00      White shark                R. Collier, GSAF   
NaN  14h00  -15h00              NaN  K.McMurray, TrackingSharks.com   
NaN          07h45              NaN  K.McMurray, TrackingSharks.com   
NaN            NaN        2 m shark                  B. Myatt, GSAF   
NaN            NaN  Tiger shark, 3m                       A .Kipper   
..             ...              ...                             ...   
NaN            NaN              NaN                             NaN   
NaN            NaN              NaN                             NaN   
NaN            NaN              NaN                             NaN   
NaN            NaN              NaN                             NaN   
NaN            NaN              NaN                             NaN   

                          pdf  \
NaN      2018.06.25-Wolfe.pdf   
NaN    2018.06.18-McNeely.pdf   
NaN     2018.06.09-Denges.pdf   
NaN  2018.06.08-Arrawarra.pdf   
NaN      2018.06.04-Ramos.pdf   
..                        ...   
NaN                       NaN   
NaN                       NaN   
NaN                       NaN   
NaN                       NaN   
NaN                       NaN   

                                          href_formula  \
NaN  http://sharkattackfile.net/spreadsheets/pdf_di...   
NaN  http://sharkattackfile.net/spreadsheets/pdf_di...   
NaN  http://sharkattackfile.net/spreadsheets/pdf_di...   
NaN  http://sharkattackfile.net/spreadsheets/pdf_di...   
NaN  http://sharkattackfile.net/spreadsheets/pdf_di...   
..                                                 ...   
NaN                                                NaN   
NaN                                                NaN   
NaN                                                NaN   
NaN                                                NaN   
NaN                                                NaN   

                                                  href case_number.1  \
NaN  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
NaN

In [9]:
projshark = attacks[['date','year','country','location','activity','fatal','type','area']]

projshark.dropna(how='all') 


date    year             country  \
NaN  25-Jun-2018  2018.0                 USA   
NaN  18-Jun-2018  2018.0                 USA   
NaN  09-Jun-2018  2018.0                 USA   
NaN  08-Jun-2018  2018.0           AUSTRALIA   
NaN  04-Jun-2018  2018.0              MEXICO   
..           ...     ...                 ...   
NaN  Before 1903     0.0           AUSTRALIA   
NaN  Before 1903     0.0           AUSTRALIA   
NaN    1900-1905     0.0                 USA   
NaN    1883-1889     0.0              PANAMA   
NaN    1845-1853     0.0  CEYLON (SRI LANKA)   

                                location      activity fatal        type  \
NaN          Oceanside, San Diego County      Paddling     N     Boating   
NaN       St. Simon Island, Glynn County      Standing     N  Unprovoked   
NaN                         Habush, Oahu       Surfing     N     Invalid   
NaN                   Arrawarra Headland       Surfing     N  Unprovoked   
NaN                             La Ticla   Free diving     N    Provoked   
..                                   ...           ...   ...         ...   
NaN                          Roebuck Bay        Diving     Y  Unprovoked   
NaN                                  NaN  Pearl diving     Y  Unprovoked   
NaN                       Ocracoke Inlet      Swimming     Y  Unprovoked   
NaN                 Panama Bay 8ºN, 79ºW           NaN     Y  Unprovoked   
NaN  Below the English fort, Trincomalee      Swimming     Y  Unprovoked   

                  area  
NaN         California  
NaN            Georgia  
NaN             Hawaii  
NaN    New South Wales  
NaN             Colima  
..                 ...  
NaN  Western Australia  
NaN  Western Australia  
NaN     North Carolina  
NaN                NaN  
NaN   Eastern Province  

[6302 rows x 8 columns]

In [10]:
projshark['country'].value_counts()

USA                       2229
AUSTRALIA                 1338
SOUTH AFRICA               579
PAPUA NEW GUINEA           134
NEW ZEALAND                128
                          ... 
MALDIVE ISLANDS              1
NICARAGUA                    1
NORTH SEA                    1
RED SEA / INDIAN OCEAN       1
CEYLON (SRI LANKA)           1
Name: country, Length: 212, dtype: int64

In [11]:
projsharkattacks = projshark[(projshark["year"] >=2000)]

projsharkattacks.year.unique()


array([2018., 2017., 2016., 2015., 2014., 2013., 2012., 2011., 2010.,
       2009., 2008., 2007., 2006., 2005., 2004., 2003., 2002., 2001.,
       2000.])

In [12]:
proj_sharkattacks = projsharkattacks[(projsharkattacks["country"] =="USA") | (projsharkattacks["country"] =="AUSTRALIA")] 
proj_sharkattacks

date    year    country                                  location  \
NaN  25-Jun-2018  2018.0        USA               Oceanside, San Diego County   
NaN  18-Jun-2018  2018.0        USA            St. Simon Island, Glynn County   
NaN  09-Jun-2018  2018.0        USA                              Habush, Oahu   
NaN  08-Jun-2018  2018.0  AUSTRALIA                        Arrawarra Headland   
NaN  03-Jun-2018  2018.0  AUSTRALIA                        Flat Rock, Ballina   
..           ...     ...        ...                                       ...   
NaN  02-Mar-2000  2000.0  AUSTRALIA   Taronga Wharf, Athol Bay, Sydney Harbor   
NaN     Mar-2000  2000.0        USA         Midnight Lump (38 miles offshore)   
NaN  21-Feb-2000  2000.0        USA          Riviera Beach, Palm Beach County   
NaN  01-Feb-2000  2000.0  AUSTRALIA  Point Sinclair, Cactus Beach near Penong   
NaN         2000  2000.0        USA                   Boca Grande, Lee County   

               activity fatal        type             area  
NaN            Paddling     N     Boating       California  
NaN            Standing     N  Unprovoked          Georgia  
NaN             Surfing     N     Invalid           Hawaii  
NaN             Surfing     N  Unprovoked  New South Wales  
NaN        Kite surfing     N  Unprovoked  New South Wales  
..                  ...   ...         ...              ...  
NaN            Swimming     N  Unprovoked  New South Wales  
NaN        Spearfishing     N  Unprovoked        Louisiana  
NaN                 NaN     N  Unprovoked          Florida  
NaN             Surfing     N  Unprovoked  South Australia  
NaN  Fishing for tarpon     N        Boat          Florida  

[1409 rows x 8 columns]

In [13]:
projsharkattacks['country'].value_counts()

USA             1014
AUSTRALIA        395
SOUTH AFRICA     137
BRAZIL            56
BAHAMAS           54
                ... 
MALDIVES           1
MALTA              1
SOMALIA            1
NEVIS              1
DIEGO GARCIA       1
Name: country, Length: 98, dtype: int64

In [14]:
def tratameno_fatal(x):
    if x == 'Y':
        return 1
    elif x == 'N':
        return 0
    else:
        return 'Unknown'

In [15]:
proj_sharkattacks = proj_sharkattacks.reset_index()

In [16]:
proj_sharkattacks['fatal'].value_counts()

N          1260
Y            53
UNKNOWN       2
Name: fatal, dtype: int64

In [17]:

proj_sharkattacks['occurrences'] = proj_sharkattacks['fatal'].map(tratameno_fatal)
proj_sharkattacks['qt']=1
proj_sharkattacks

index         date    year    country  \
0       NaN  25-Jun-2018  2018.0        USA   
1       NaN  18-Jun-2018  2018.0        USA   
2       NaN  09-Jun-2018  2018.0        USA   
3       NaN  08-Jun-2018  2018.0  AUSTRALIA   
4       NaN  03-Jun-2018  2018.0  AUSTRALIA   
...     ...          ...     ...        ...   
1404    NaN  02-Mar-2000  2000.0  AUSTRALIA   
1405    NaN     Mar-2000  2000.0        USA   
1406    NaN  21-Feb-2000  2000.0        USA   
1407    NaN  01-Feb-2000  2000.0  AUSTRALIA   
1408    NaN         2000  2000.0        USA   

                                      location            activity fatal  \
0                  Oceanside, San Diego County            Paddling     N   
1               St. Simon Island, Glynn County            Standing     N   
2                                 Habush, Oahu             Surfing     N   
3                           Arrawarra Headland             Surfing     N   
4                           Flat Rock, Ballina        Kite surfing     N   
...                                        ...                 ...   ...   
1404   Taronga Wharf, Athol Bay, Sydney Harbor            Swimming     N   
1405         Midnight Lump (38 miles offshore)        Spearfishing     N   
1406          Riviera Beach, Palm Beach County                 NaN     N   
1407  Point Sinclair, Cactus Beach near Penong             Surfing     N   
1408                   Boca Grande, Lee County  Fishing for tarpon     N   

            type             area occurrences  qt  
0        Boating       California           0   1  
1     Unprovoked          Georgia           0   1  
2        Invalid           Hawaii           0   1  
3     Unprovoked  New South Wales           0   1  
4     Unprovoked  New South Wales           0   1  
...          ...              ...         ...  ..  
1404  Unprovoked  New South Wales           0   1  
1405  Unprovoked        Louisiana           0   1  
1406  Unprovoked          Florida           0   1  
1407  Unprovoked  South Australia           0   1  
1408        Boat          Florida           0   1  

[1409 rows x 11 columns]

In [18]:
sharkattacksproj = proj_sharkattacks[proj_sharkattacks['occurrences'] != 'Unknown']
sharkattacksproj

index         date    year    country  \
0       NaN  25-Jun-2018  2018.0        USA   
1       NaN  18-Jun-2018  2018.0        USA   
2       NaN  09-Jun-2018  2018.0        USA   
3       NaN  08-Jun-2018  2018.0  AUSTRALIA   
4       NaN  03-Jun-2018  2018.0  AUSTRALIA   
...     ...          ...     ...        ...   
1404    NaN  02-Mar-2000  2000.0  AUSTRALIA   
1405    NaN     Mar-2000  2000.0        USA   
1406    NaN  21-Feb-2000  2000.0        USA   
1407    NaN  01-Feb-2000  2000.0  AUSTRALIA   
1408    NaN         2000  2000.0        USA   

                                      location            activity fatal  \
0                  Oceanside, San Diego County            Paddling     N   
1               St. Simon Island, Glynn County            Standing     N   
2                                 Habush, Oahu             Surfing     N   
3                           Arrawarra Headland             Surfing     N   
4                           Flat Rock, Ballina        Kite surfing     N   
...                                        ...                 ...   ...   
1404   Taronga Wharf, Athol Bay, Sydney Harbor            Swimming     N   
1405         Midnight Lump (38 miles offshore)        Spearfishing     N   
1406          Riviera Beach, Palm Beach County                 NaN     N   
1407  Point Sinclair, Cactus Beach near Penong             Surfing     N   
1408                   Boca Grande, Lee County  Fishing for tarpon     N   

            type             area occurrences  qt  
0        Boating       California           0   1  
1     Unprovoked          Georgia           0   1  
2        Invalid           Hawaii           0   1  
3     Unprovoked  New South Wales           0   1  
4     Unprovoked  New South Wales           0   1  
...          ...              ...         ...  ..  
1404  Unprovoked  New South Wales           0   1  
1405  Unprovoked        Louisiana           0   1  
1406  Unprovoked          Florida           0   1  
1407  Unprovoked  South Australia           0   1  
1408        Boat          Florida           0   1  

[1313 rows x 11 columns]

In [19]:
sharkattacksproj['area'].value_counts()


Florida                                 515
New South Wales                         145
Hawaii                                  120
California                              103
Western Australia                        87
South Carolina                           68
Queensland                               60
North Carolina                           55
Texas                                    30
South Australia                          28
Victoria                                 27
Oregon                                   14
Alabama                                   9
Massachusetts                             6
Northern Territory                        5
Tasmania                                  5
New Jersey                                4
Westerm Australia                         3
Georgia                                   3
Virginia                                  3
Louisiana                                 3
Puerto Rico                               2
Palmyra Atoll                   

In [20]:
sharkattacksproj['country'].value_counts()

USA          951
AUSTRALIA    362
Name: country, dtype: int64

In [21]:
year_ = pd.qcut(sharkattacksproj.year, 3)

year_.unique()

[(2013.0, 2018.0], (2007.0, 2013.0], (1999.999, 2007.0]]
Categories (3, interval[float64, right]): [(1999.999, 2007.0] < (2007.0, 2013.0] < (2013.0, 2018.0]]

In [22]:
sharkattacksproj = sharkattacksproj.reset_index()

In [23]:
names = ['2000 a 2007', '2008 a 2013', '2014 a 2018']
years = pd.qcut(sharkattacksproj.year, 3,labels=names) 
sharkattacksproj['years']=years
sharkattacksproj.groupby(['years','country',])[['fatal']].count()

fatal
years       country         
2000 a 2007 AUSTRALIA    111
            USA          400
2008 a 2013 AUSTRALIA    120
            USA          295
2014 a 2018 AUSTRALIA    131
            USA          256

In [25]:
sharkattacks_perc = sharkattacksproj.groupby(['years','country'])[['occurrences']].count().sort_values(by=['years','occurrences'], ascending=False)

In [26]:
shark_attacks_perc = round(sharkattacks_perc[['occurrences']]*100 / sharkattacks_perc.groupby('years')[['occurrences']].sum(),2).astype('str')+'%'
sharkattacks_perc['percente']=shark_attacks_perc['occurrences']
sharkattacks_perc

occurrences percente
years       country                        
2014 a 2018 USA                256   66.15%
            AUSTRALIA          131   33.85%
2008 a 2013 USA                295   71.08%
            AUSTRALIA          120   28.92%
2000 a 2007 USA                400   78.28%
            AUSTRALIA          111   21.72%

In [27]:
sharkattacks_country = sharkattacksproj.groupby('country')[['occurrences']].count().sort_values(by=['occurrences'], ascending=False)

In [28]:
sharkattacks_country

occurrences
country               
USA                951
AUSTRALIA          362

In [29]:
sharkattacks_perc2 = sharkattacksproj.groupby(['years','fatal','country'])[['occurrences']].count().sort_values(by=['years','occurrences'], ascending=False)

In [30]:
shark_attacks_perc2 = round(sharkattacks_perc2[['occurrences']]*100 / sharkattacks_perc2.groupby('years')[['occurrences']].sum(),2).astype('str')+'%'
sharkattacks_perc2['percente']=shark_attacks_perc2['occurrences']
sharkattacks_perc2

occurrences percente
years       fatal country                        
2014 a 2018 N     USA                255   65.89%
                  AUSTRALIA          121   31.27%
            Y     AUSTRALIA           10    2.58%
                  USA                  1    0.26%
2008 a 2013 N     USA                287   69.16%
                  AUSTRALIA          109   26.27%
            Y     AUSTRALIA           11    2.65%
                  USA                  8    1.93%
2000 a 2007 N     USA                389   76.13%
                  AUSTRALIA           99   19.37%
            Y     AUSTRALIA           12    2.35%
                  USA                 11    2.15%

In [31]:
USA = sharkattacksproj[sharkattacksproj['country'] != 'AUSTRALIA']   
AUSTRALIA = sharkattacksproj[sharkattacksproj['country'] != 'USA']  
pd.pivot_table(USA, index = 'type', columns = ['country','years'],values = 'occurrences', aggfunc='count',dropna =True, fill_value=0, margins =True,sort=True ).sort_values(by=['All'], ascending=False)


country              USA                          All
years        2000 a 2007 2008 a 2013 2014 a 2018     
type                                                 
All                  400         295         256  951
Unprovoked           359         261         223  843
Provoked              28          21          13   62
Boating                4           9          19   32
Invalid                4           2           1    7
Boat                   4           1           0    5
Sea Disaster           1           1           0    2

In [32]:
  
pd.pivot_table(AUSTRALIA, index = 'type', columns = ['country','years'],values = 'occurrences', aggfunc='count',dropna =True, fill_value=0, margins =True,sort=True ).sort_values(by=['All'], ascending=False)

country        AUSTRALIA                          All
years        2000 a 2007 2008 a 2013 2014 a 2018     
type                                                 
All                  111         120         131  362
Unprovoked            88          95         101  284
Provoked               8          11          15   34
Boating                9          10           9   28
Boat                   5           4           0    9
Invalid                1           0           4    5
Questionable           0           0           2    2

Florida                                 515
New South Wales                         145
Hawaii                                  120
California                              103
Western Australia                        87
South Carolina                           68
Queensland                               60
North Carolina                           55
Texas                                    30
South Australia                          28
Victoria                                 27
Oregon                                   14

In [33]:
areaprojaust = sharkattacksproj[(sharkattacksproj["area"] =="New South Wales") |(sharkattacksproj["area"] =="Western Australia")] 

areaprojusa = sharkattacksproj[(sharkattacksproj["area"] =="Florida")|(sharkattacksproj["area"] =="Hawaii")|(sharkattacksproj["area"] =="California")] 
areaprojusa.area.value_counts()

Florida       515
Hawaii        120
California    103
Name: area, dtype: int64

In [34]:
#pd.pivot_table(sharkattacksproj, index = 'type', columns = ['country','years'],values = 'occurrences', aggfunc='count',dropna =True, fill_value=0, margins =True,sort=True ).sort_values(by=['All'], ascending=False)

pd.pivot_table(areaprojaust, index = 'area', columns = ['country','years'],values = 'occurrences', aggfunc='count', margins=True, sort=True).sort_values(by=['All'], ascending=False)

country             AUSTRALIA                          All
years             2000 a 2007 2008 a 2013 2014 a 2018     
area                                                      
All                        59          86          87  232
New South Wales            39          54          52  145
Western Australia          20          32          35   87

In [35]:
pd.pivot_table(areaprojusa, index = 'area', columns = ['country','years'],values = 'occurrences', aggfunc='count', margins=True,sort=True).sort_values(by=['All'], ascending=False)

country            USA                          All
years      2000 a 2007 2008 a 2013 2014 a 2018     
area                                               
All                315         226         197  738
Florida            241         147         127  515
Hawaii              41          44          35  120
California          33          35          35  103

In [49]:
#areaprojfim = sharkattacksproj[(sharkattacksproj["area"] =="Florida")]
#areaprojfim.area.value_counts()
areaprojflorida = sharkattacksproj[(sharkattacksproj["area"] =="Florida")]
areaprojhawaii = sharkattacksproj[(sharkattacksproj["area"] =="Hawaii")]
areaprojcalifornia = sharkattacksproj[(sharkattacksproj["area"] =="California")]
pd.pivot_table(areaprojflorida, index = 'location', columns = ['area','years'],values = 'occurrences', aggfunc='sum', margins=True,sort=True).sort_values(by=['All'], ascending=False)

area                                                   Florida              \
years                                              2000 a 2007 2008 a 2013   
location                                                                     
All                                                          4           1   
Boca Ciega Bay, Tampa, Pinellas County                       1           0   
Stuart, Martin County                                        0           1   
Palm Beach?                                                  1           0   
2 miles off Pompano Beach, Broward County                    1           0   
...                                                        ...         ...   
Islamorada Founder's Park, Plantation Key, Monr...           0           0   
Islamorada, Monroe County                                    0           0   
Jacksonville                                                 0           0   
Jacksonville Beach, Duval County                             0           0   
Melbourne Beach, Brevard County                              0           0   

area                                                           All  
years                                              2014 a 2018      
location                                                            
All                                                          0   5  
Boca Ciega Bay, Tampa, Pinellas County                       0   1  
Stuart, Martin County                                        0   1  
Palm Beach?                                                  0   1  
2 miles off Pompano Beach, Broward County                    0   1  
...                                                        ...  ..  
Islamorada Founder's Park, Plantation Key, Monr...           0   0  
Islamorada, Monroe County                                    0   0  
Jacksonville                                                 0   0  
Jacksonville Beach, Duval County                             0   0  
Melbourne Beach, Brevard County                              0   0  

[249 rows x 4 columns]

In [37]:
areaprojfloridal = sharkattacksproj[(sharkattacksproj["area"] =="Florida") & (sharkattacksproj["location"] =="New Smyrna Beach, Volusia County")]

pd.pivot_table(areaprojfloridal, index = ['activity'], columns = ['area','years'],values = 'occurrences', aggfunc='count', margins=True,sort=True).sort_values(by=['All'], ascending=False)

area                            Florida                          All
years                       2000 a 2007 2008 a 2013 2014 a 2018     
activity                                                            
All                                  58          34          27  119
Surfing                              40          26          21   87
Wading                                4           4           2   10
Swimming                              4           0           1    5
Boogie boarding                       2           1           0    3
Body boarding                         1           0           1    2
Body surfing                          1           0           1    2
Boogie Boarding                       2           0           0    2
Playing on a sandbar                  1           0           0    1
Playing soccer in the water           1           0           0    1
Standing                              1           0           0    1
Surfing                               0           1           0    1
Surfng                                0           0           1    1
Swimming or boogie boarding           0           1           0    1
Swimming, towing surfboard            0           1           0    1
Wading?                               1           0           0    1

In [38]:
areaprojfim2 = sharkattacksproj[(sharkattacksproj["area"] =="New South Wales")]
areaprojfim2.area.value_counts()

pd.pivot_table(areaprojfim2, index = ['area','location'], columns = 'years',values = 'occurrences', aggfunc='count', margins =True,dropna =True, sort=True ).sort_values(by=['area','All'], ascending=False)

years                                         2000 a 2007  2008 a 2013  \
area            location                                                 
New South Wales Avoca Beach                             0            1   
                Sharpes Beach, Ballina                  0            0   
                Broken Head                             0            2   
                Bronte Beach                            2            0   
                Byron Bay                               2            0   
...                                                   ...          ...   
                Wallabi Point                           0            0   
                Wategos Beach, Byon Bay                 1            0   
                Windang,       Lake Illawara            0            1   
                Wollongong                              1            0   
All                                                    38           54   

years                                         2014 a 2018  All  
area            location                                        
New South Wales Avoca Beach                             2    3  
                Sharpes Beach, Ballina                  3    3  
                Broken Head                             0    2  
                Bronte Beach                            0    2  
                Byron Bay                               0    2  
...                                                   ...  ...  
                Wallabi Point                           1    1  
                Wategos Beach, Byon Bay                 0    1  
                Windang,       Lake Illawara            0    1  
                Wollongong                              0    1  
All                                                    52  144  

[126 rows x 4 columns]

In [39]:
areaprojfim2 = sharkattacksproj[(sharkattacksproj["area"] =="New South Wales")]
areaprojfim2.area.value_counts()

pd.pivot_table(areaprojfim2, index = ['area','activity'], columns = 'years',values = 'occurrences', aggfunc='count', margins =True,dropna =True, sort=True ).sort_values(by=['area','All'], ascending=False)

years                                                               2000 a 2007  \
area            activity                                                          
New South Wales Surfing                                                      18   
                Swimming                                                      4   
                Fishing                                                       1   
                Body boarding                                                 0   
                Diving                                                        3   
                Kayaking                                                      2   
                Rowing                                                        2   
                Snorkeling                                                    2   
                Surf skiing                                                   0   
                Boogie Boarding                                               1   
                Paddle boarding                                               0   
                Spearfishing                                                  1   
                Standing                                                      2   
                Surf-skiing                                                   1   
                Body boarding                                                 0   
                Bodysurfing                                                   0   
                Diving, but on the surface when bitten by the s...            0   
                Fell onto dead shark                                          1   
                Filming                                                       0   
                Fishing                                                       0   
                Free diving                                                   0   
                Kite surfing                                                  0   
                Paddle-boarding                                               0   
                Scuba diving                                                  0   
                Surf fishing                                                  0   
                Teasing a shark                                               0   
                Wakeboarding                                                  0   
                Walking                                                       0   
                Windsurfing                                                   0   
                Wrangling a shark                                             0   
All                                                                          38   

years                                                               2008 a 2013  \
area            activity                                                          
New South Wales Surfing                                                      32   
                Swimming                                                      2   
                Fishing                                                       4   
                Body boarding                                                 3   
                Diving                                                        0   
                Kayaking                                                      1   
                Rowing                                                        1   
                Snorkeling                                                    1   
                Surf skiing                                                   2   
                Boogie Boarding                                               1   
                Paddle boarding                                               1   
                Spearfishing                                                  1   
                Standing                                                      0   
               

In [40]:
areaprojfim3 = sharkattacksproj[(sharkattacksproj["area"] =="Western Australia")]
areaprojfim3.area.value_counts()

pd.pivot_table(areaprojfim3, index = ['area','location'], columns = 'years',values = 'occurrences', aggfunc='count', margins =True,dropna =True, sort=True ).sort_values(by=['area','All'], ascending=False)

years                                            2000 a 2007  2008 a 2013  \
area              location                                                  
Western Australia Bunbury                                  1            0   
                  Coral Bay                                0            1   
                  Cottesloe Beach, Perth                   2            0   
                  Gracetown                                0            1   
                  Mullaloo Beach, Perth                    0            2   
...                                                      ...          ...   
                  Waterman's Bay                           0            0   
                  Wedge Island                             0            1   
                  Wharton Beach                            1            0   
                  Wreck Point, Abrolhos Islands            1            0   
All                                                       20           31   

years                                            2014 a 2018  All  
area              location                                         
Western Australia Bunbury                                  1    2  
                  Coral Bay                                1    2  
                  Cottesloe Beach, Perth                   0    2  
                  Gracetown                                1    2  
                  Mullaloo Beach, Perth                    0    2  
...                                                      ...  ...  
                  Waterman's Bay                           1    1  
                  Wedge Island                             0    1  
                  Wharton Beach                            0    1  
                  Wreck Point, Abrolhos Islands            0    1  
All                                                       35   86  

[78 rows x 4 columns]